### PR TITLE
fix(build): turn c_src dependencies into git submodules

### PR DESCRIPTION
**Pull request by keynslug**
Also make fetching source repos simpler and more foolproof by turning them into git submodules.

### CODE REVIEW

**Review by zmstone — COMMENTED** [review]
maybe git submodule.

**Comment by keynslug** [comment]
Looks like it was some weird Make quirk where it ignored variable definition if defined `define smth =` (not surprising given how ancient MacOS Make is).

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[submodule "c_src/libs/c_reusable_data_structures"]
+	path = c_src/libs/c_reusable_data_structures
+	url = https://github.com/emqx/c_reusable_data_structures.git
+	branch = master
```
Comment (zmstone) [inline .gitmodules:4]: still, better to pin a tag instead

Comment (keynslug) [reply .gitmodules:4]: But it's just a tracking branch, the submodule itself is pinned by commit hash in the git object itself.
https://github.com/emqx/jq/pull/49/files/9d87fffa3e07085ca4b71fe7a5df8cf32cde1450#diff-13821eeef0636f9fc5f8a0677131978922278043a982629aba9869556721fe18

**Review by zmstone — APPROVED** [review]

**Pull request merged**